rskj-core/src/test/java/co/rsk/peg/BridgeSupportIT.java

```diff
@@ -192,17 +199,114 @@ void testInitialChainHeadWithoutBtcCheckpoints() throws Exception {
+        StoredBlock checkpoint = manager.getCheckpointBefore(time);
+
+        assertEquals(checkpoint.getHeight(), bridgeSupport.getBtcBlockchainBestChainHeight());
+    }
+
+    @Test
+    void testInitialChainHeadWithBtcCheckpoints_whenCheckpointsWith32BytesChainWork_before_RSKIP454_shouldFail() throws Exception {
+        var arrowhead631 = ActivationConfigsForTest.arrowhead631().forBlock(0);
+        BridgeConstants bridgeConstants = BridgeMainNetConstants.getInstance();
+        String checkpointToCreate = "/rskbitcoincheckpoints/" + bridgeConstants.getBtcParams().getId() + ".checkpoints";
+
+        Path target = Paths.get(getClass().getResource(checkpointToCreate).getPath());
+
+        Path source = Paths.get(getClass().getResource("/checkpoints/32-byte-chainwork.production.checkpoints").getPath());
+        Files.copy(source, target, StandardCopyOption.REPLACE_EXISTING);
+
+        BridgeSupport bridgeSupport = arrangeBridgeSupport(bridgeConstants, arrowhead631);
+
+        Assertions.assertThrows(IllegalArgumentException.class, bridgeSupport::getBtcBlockchainBestChainHeight, "The given number does not fit in 12");
+    }
+
+    @ParameterizedTest
+    @CsvSource(
+        {
+            "12-byte-chainwork.production.checkpoints",
+            "12-byte-chainwork.production.checkpoints",
+            "12-byte-chainwork-mix-format.production.checkpoints"
+        }
+    )
+    void testInitialChainHeadWithBtcCheckpoints_whenCheckpointsWith32BytesChainWork_after_RSKIP454_ok(
+        String checkpointFileName) throws Exception {
+        var lovell = ActivationConfigsForTest.lovell700().forBlock(0);
+        BridgeConstants bridgeConstants = BridgeMainNetConstants.getInstance();
+        String checkpointToCreate = "/rskbitcoincheckpoints/" + bridgeConstants.getBtcParams().getId() + ".checkpoints";
+
+        Path target = Paths.get(getClass().getResource(checkpointToCreate).getPath());
+
+        Path source = Paths.get(getClass().getResource("/checkpoints/" + checkpointFileName).getPath());
+        Files.copy(source, target, StandardCopyOption.REPLACE_EXISTING);
+
+        BridgeSupport bridgeSupport = arrangeBridgeSupport(bridgeConstants, lovell);
+
+        // Force instantiation of blockstore
+        bridgeSupport.getBtcBlockchainBestChainHeight();
+
+        InputStream checkpointsStream = bridgeSupport.getCheckPoints();
+        CheckpointManager manager = new CheckpointManager(bridgeConstants.getBtcParams(), checkpointsStream);
+        long time = bridgeSupport.getActiveFederation().getCreationTime().toEpochMilli() - 604800L; // The magic number is a substraction CheckpointManager does when getting the checkpoints.
+        StoredBlock checkpoint = manager.getCheckpointBefore(time);
+
+        assertEquals(checkpoint.getHeight(), bridgeSupport.getBtcBlockchainBestChainHeight());
+    }
+
+    private BridgeSupport arrangeBridgeSupport(BridgeConstants bridgeTestNetConstants,
```

Comment:
not a big fan of this. Can't we set the `bridgeSupport` in the general setup?

Reply:
I need to build the `bridgeSupport` using different bridgeConstants and different activations depending on the use case. This is why I don't put it there, but I'm reorganizing the tests, and let's see if you find it easier to follow.